### DISCUSSION TITLE
Is there a fundamental reason that the width cannot be set on Windows?

### DISCUSSION THREAD

**Discussion by wsw70** [ncruces/zenity#57] [Q&A] [answered]
I believe there is :) but I was wondering if this could be addressed one way or another? (I would like to make my window 80% screen wide)

**Comment by ncruces** [comment]
No, none:
- for all custom dialogs (Calendar, Entry, List, Progress) it's just additional layout work;
- for Error/Info/Warning/Question it'd be more of a hack, but probably doable;
- other dialogs don't have it even on Linux.

**Reply by wsw70** [reply]
Thank you. I was looking at `Entry`-  you mean that the layout is handcrafted (what goes where, exactly), yes? In the sense that setting a larger width would not rearrange all the components?

**Reply by ncruces** [reply, marked as answer]
Yes:
https://github.com/ncruces/zenity/blob/ee60339b2dd4cb5b1c1f9a5b2401fff962ca5918/entry_windows.go#L121-L139

**Reply by wsw70** [reply]
Thank you, I should have had a look at the source indeed. I will have a closer look to see if I can hack a configurable width, at least for my case.

Your library is really great.

**Reply by ncruces** [reply]
I don't blame you. The source code is a bit daunting. It's what I get for trying to do it all with limited (runtime) dependencies.

If you come up with something nice, maybe do a PR.

Thanks!
